compiler/qsc_eval/src/backend.rs

```diff
@@ -56,7 +60,24 @@ impl SparseSim {
     }
 }
 
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub struct SparseSimResult(bool);
+
+impl From<val::Result> for SparseSimResult {
+    fn from(r: val::Result) -> Self {
+        Self(r.into())
+    }
+}
+
+impl From<SparseSimResult> for val::Result {
+    fn from(r: SparseSimResult) -> Self {
+        r.0.into()
+    }
+}
+
 impl Backend for SparseSim {
+    type ResultType = SparseSimResult;
```

Comment:
Couldn't this be bool, since there's an Into for bool to val::Result?

Reply:
Huh, I think you are right. This went through several refactorings so I didn't notice that.